projects/Button_ADC/src/main.c

```diff
@@ -0,0 +1,53 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress button_address = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioAddress adc_address = { .port = GPIO_PORT_A, .pin = 6 };
+
+static GpioSettings adc_settings = {
+  .direction = GPIO_DIR_IN,
+  .resistor = GPIO_RES_NONE,
+  .alt_function = GPIO_ALTFN_ANALOG,
+};
+
+static InterruptSettings interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
+
+static AdcChannel adc_pinA6_channel = ADC_CHANNEL_6;
```

Comment:
Nit: our variable names are generally all lowercase and we don't use lowerCamelCase at all (so this should rather be `adc_pin_a6_channel`). Also, why are you initializing it to `ADC_CHANNEL_6`?